nim/tests/test_routes.py

```diff
@@ -0,0 +1,149 @@
+import json
+
+import pytest
+
+from nim.game import (
+    DEFAULT_MIN,
+    DEFAULT_MAX,
+    DEFAULT_PILES
+)
+
+from nim.app import app
+
+
+@pytest.fixture
+def client_app():
+    return app.test_client()
+
+
+def assert_new_game_success(passed_client, req):
+    """
+    Asserts post to /new is 200
+    & state is not empty
+    Checks piles is in line with req &
+    none in response lower than mix or
+    higher than max
+    :param passed_client: test_client
+    :param req: Desired game conditions
+    """
+    response = passed_client.post(
+        '/new',
+        content_type='application/json',
+        data=req
+    )
```

Comment:
Below, you may want to add some inline comments below, maybe some empty lines too so it doesn't look scary to read.